python/MooseDocs/extensions/MooseImageFile.py

```diff
@@ -0,0 +1,103 @@
+import re
+import os
+from markdown.util import etree
+import logging
+log = logging.getLogger(__name__)
+
+import MooseDocs
+from markdown.inlinepatterns import Pattern
+from MooseCommonExtension import MooseCommonExtension
+import utils
+
+class MooseImageFile(MooseCommonExtension, Pattern):
+    """
+    Markdown extension for handling images.
+    """
+
+    # Find !image /path/to/file attribute=setting
+    RE = r'!image\s+(.*?)\s+(.*)'
+    SETTINGS_RE = re.compile("([^\s=]+)=(.*?)(?=(?:\s[^\s=]+=|$))")
```

Comment:
Yeah, that same re is a class variable in `MooseCarousel`. So that should probably be moved to `MooseCommonExtension` as well as the `parseOptions` function.